servers/rendering/renderer_rd/renderer_compositor_rd.cpp

```diff
@@ -155,6 +156,45 @@ void RendererCompositorRD::finalize() {
 RendererCompositorRD *RendererCompositorRD::singleton = nullptr;
 
 RendererCompositorRD::RendererCompositorRD() {
+	{
+		String shader_cache_dir = Engine::get_singleton()->get_shader_cache_path();
+		if (shader_cache_dir == String()) {
+			shader_cache_dir = "user://";
+		}
+		DirAccessRef da = DirAccess::open(shader_cache_dir);
+		if (!da) {
+			ERR_PRINT("Can't create shader cache dir, no shader caching will happen: " + shader_cache_dir);
+		} else {
+			Error err = da->change_dir("shader_cache");
```

Comment:
When testing this PR locally, I noticed the shader cache was created as a visible folder in my project directory. Is the path specified as a fallback correct, or should it be `.godot/shader_cache`?

Reply:
It should be .godot/shader_cache when running in the editor, I have no idea why it put it in your project folder root.